### PR TITLE
chore: add non-sso-tab-default-bot template (js)

### PR DESCRIPTION
**Pull request by yuqizhou77**
- Add a new template for both tab and bot capability
- **This template won't be shown in UI for user to select**, but just to help users from Developer Portal who has an app created before with both tab and bot capability selected.
- Verified in local and dev.
image
image

### CODE REVIEW

File: templates/scenarios/js/non-sso-tab-default-bot/.vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        },
+        {
+            // Start the local tunnel service to forward public ngrok URL to local port and inspect traffic.
+            // See https://aka.ms/teamsfx-local-tunnel-task to know the details and how to customize the args.
+            "label": "Start local tunnel",
+            "type": "teamsfx",
+            "command": "debug-start-local-tunnel",
+            "args": {
+                "ngrokArgs": "http 3978 --log=stdout --log-format=logfmt",
+                "env": "local"
```
Comment (kuojianlu) [inline templates/scenarios/js/non-sso-tab-default-bot/.vscode/tasks.json:45]: Please apply the latest change in https://github.com/OfficeDev/TeamsFx/pull/6889.

File: templates/scenarios/js/non-sso-tab-default-bot/teamsfx/app.local.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      name: bot
+    # Output: following environment variable will be persisted in current environment's .env file.
+    # BOT_ID: the AAD app client id created for bot
+    # SECRET_BOT_PASSWORD: the AAD app client secret created for bot
+
+  - uses: botFramework/createOrUpdateBot # Create or update the bot registration on dev.botframework.com
+    with:
+      botId: ${{BOT_ID}}
+      name: bot
+      messagingEndpoint: ${{TUNNEL_ENDPOINT}}/api/messages
```
Comment (kuojianlu) [inline templates/scenarios/js/non-sso-tab-default-bot/teamsfx/app.local.yml:22]:
```suggestion
      messagingEndpoint: ${{BOT_ENDPOINT}}/api/messages
```

File: templates/scenarios/js/non-sso-tab-default-bot/teamsfx/app.local.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+configureApp:
+  - uses: env/generate # Generate env to .env file
+    with:
+      envs:
+        TAB_DOMAIN: localhost:53000
+        TAB_ENDPOINT: https://localhost:53000
+  - uses: teamsApp/validate
+    env: 
+      BOT_DOMAIN: ${{TUNNEL_DOMAIN}}
```
Comment (kuojianlu) [inline templates/scenarios/js/non-sso-tab-default-bot/teamsfx/app.local.yml:33]: Remove.

File: templates/scenarios/js/non-sso-tab-default-bot/teamsfx/app.local.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    with:
+      manifestTemplatePath: ./appPackage/manifest.template.json # Path to manifest template
+  - uses: teamsApp/createAppPackage # Build Teams app package with latest env value
+    env: 
+      BOT_DOMAIN: ${{TUNNEL_DOMAIN}}
```
Comment (kuojianlu) [inline templates/scenarios/js/non-sso-tab-default-bot/teamsfx/app.local.yml:38]: Remove.

File: templates/scenarios/js/non-sso-tab-default-bot/teamsfx/script/run.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+async function run() {
+  const args = process.argv.slice(2);
+
+  if (args.length !== 3) {
+    console.log(`Usage: node ${__filename} [project path] [env path].`);
+    process.exit(1);
+  }
+
+  const capability = args[2];
+  if (capability != "tab" && capability != "bot") {
```
Comment (kuojianlu) [inline templates/scenarios/js/non-sso-tab-default-bot/teamsfx/script/run.js:15]: Better to use two run scripts, e.g. `run.tab.js` and `run.bot.js`.

File: templates/scenarios/js/non-sso-tab-default-bot/tab/package.json.tpl [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        "@microsoft/teamsfx": "^2.0.0",
+        "@microsoft/teamsfx-react": "^2.0.0",
+        "axios": "^0.21.1",
+        "react": "^16.14.0",
+        "react-dom": "^16.14.0",
+        "react-router-dom": "^5.1.2",
+        "react-scripts": "^5.0.1"
+    },
+    "devDependencies": {
+        "@microsoft/teamsfx-run-utils": "alpha"
```
Comment (kuojianlu) [inline templates/scenarios/js/non-sso-tab-default-bot/tab/package.json.tpl:17]: Can be removed?

File: templates/scenarios/js/non-sso-tab-default-bot/package.json.tpl [hunk 7]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "{%appName%}",
+    "version": "0.0.1",
+    "description": "",
+    "author": "",
+    "scripts": {
+        "test": "echo \"Error: no test specified\" && exit 1",
+        "install:bot": "cd bot && npm install",
+        "install:tab": "cd tab && npm install",
+        "install": "npm run install:bot & npm run install:tab",
```
Comment (kuojianlu) [inline templates/scenarios/js/non-sso-tab-default-bot/package.json.tpl:10]: Can consider to run in parallel.

Comment (yuqizhou77) [reply templates/scenarios/js/non-sso-tab-default-bot/package.json.tpl:10]: thought & means running in parallel? Please correct me if I was wrong :)

**Review by hund030 — DISMISSED** [review]
Ok for me, as long as local debug works fine.

File: templates/scenarios/js/non-sso-tab-default-bot/package.json.tpl (hunk 7 above)
Comment (kuojianlu) [reply templates/scenarios/js/non-sso-tab-default-bot/package.json.tpl:10]: `&` will run the commands in sequence, not in parallel. Can use some package like `concurrently`.

**Review by kuojianlu — APPROVED** [review]

**Review by kuojianlu — DISMISSED** [review]

**Review by hund030 — DISMISSED** [review]

**Review by yukun-dong — DISMISSED** [review]

**Review by hund030 — APPROVED** [review]

**Pull request merged**